server/proxy/server.js

```diff
@@ -0,0 +1,61 @@
+const client = require('socket.io-client')
+const express = require('express')
+const bodyParser = require('body-parser');
+const cookieParser = require('cookie-parser')
+const cors = require('cors');
+const axios = require('axios')
+const app = express()
+
+let socket
+let cards;
+
+app.use(bodyParser.json());
+app.use(cors());
+app.use(cookieParser());
+
+app.get('/connect-and-play', (req, res) => {
+    const url = req.query.url;
+
+    socket = client(url);
+
+    const instance = axios.create({
+        baseURL: url,
+        timeout: 500,
+    });
+    
+    socket.on('connect', () => {
+        instance.post('auth/login', { 
+            username: "halfling" 
+        })
+        .then(() => {
```

Comment:
Can be a one-liner. Not an issue, just think `return` is kind of ugly in this context.

Reply:
@scottlogic-alex suggested using return to chain them all together and have the same catch block I think

Reply:
Ronan's not talking about the topology of the Promise chain.
he's saying your arrow function could be simplifed

Reply:
If you don't have curly braces, it's an implicit `return`.

Reply:
```js
.then(() => instance.get('api/cards'))
```
or if you're an absolute madman you could get rid of the arrow function entirely (don't do this):
```js
.then(instance.get.bind(null, 'api/cards'))
```